File: clients/dart/lib/api/leaderboard_api.dart
part of api;


class LeaderboardApi {
  String basePath = "https://localhost/api/v1";
  ApiClient apiClient = ApiClient.defaultApiClient;

  LeaderboardApi([ApiClient apiClient]) {
    if (apiClient != null) {
      this.apiClient = apiClient;
    }
  }

  
  /// Get current leaderboard.
  ///
  /// 
  Future<List<Leaderboard>> leaderboardGet(String method) {
    Object postBody = null;
    

    // create path and map variables
    String path = "/leaderboard".replaceAll("{format}","json");

    // query params
    Map<String, String> queryParams = {};
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if("null" != method)
      queryParams["method"] = method is List ? method.join(',') : method;
    
    

    List<String> contentTypes = ["application/json","application/x-www-form-urlencoded"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
      
    }

    return apiClient.invokeAPI(basePath, path, 'GET', queryParams, postBody, headerParams, formParams, contentType, authNames).then((response) {
      if(response.statusCode >= 400) {
        throw new ApiException(response.statusCode, response.body);
      }
      else if(response.body != null){
        return ApiClient.deserialize(response.body, Leaderboard);
      }
      else {
        return null;
      }
    });
  }
  
}

File: clients/ruby/lib/swagger_client/api/leaderboard_api.rb
require "uri"

module SwaggerClient
  class LeaderboardApi
    attr_accessor :api_client

    def initialize(api_client = nil)
      @api_client = api_client || Configuration.api_client
    end

    # Get current leaderboard.
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :method Ranking type. Options: \&quot;notional\&quot;, \&quot;ROE\&quot;
    # @return [Array<Leaderboard>]
    def leaderboard_get(opts = {})
      if Configuration.debugging
        Configuration.logger.debug "Calling API: LeaderboardApi#leaderboard_get ..."
      end
      
      # resource path
      path = "/leaderboard".sub('{format}','json')

      # query parameters
      query_params = {}
      query_params[:'method'] = opts[:'method'] if opts[:'method']

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json', 'application/xml', 'text/xml', 'application/javascript', 'text/javascript']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json', 'application/x-www-form-urlencoded']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}

      # http body (model)
      post_body = nil
      

      auth_names = []
      result = @api_client.call_api(:GET, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<Leaderboard>')
      if Configuration.debugging
        Configuration.logger.debug "API called: LeaderboardApi#leaderboard_get. Result: #{result.inspect}"
      end
      return result
    end
  end
end




